cumulusci/tasks/push/push_api.py

```diff
@@ -265,12 +268,25 @@ def __init__(self, sf, logger, lazy=None, default_where=None, batch_size=None):
             batch_size = 200
         self.batch_size = batch_size
 
-    def return_query_records(self, query):
-        res = self.sf.query_all(query)
-        if res["totalSize"] > 0:
-            return res["records"]
+        self.bulk = bulk
+
+    def return_query_records(self, query, field_names=None, sobject=None):
+        res = []
+        if self.bulk and field_names and sobject:
+            step = BulkApiQueryOperation(
+                sobject=sobject, api_options={}, context=self, query=query
+            )
+            step.query()
+            if len(list(step.get_results())) > 0:
+                for query_result in list(step.get_results()):
+                    res += [dict(zip(query_result.keys(), query_result.values()))]
```

Comment:
Keep in mind that the way this is currently implemented, we are loading all the results into memory at once rather than processing them in batches. I guess we'll see if that works, but at some point we may need to rewrite it.